src/components/MermaidDash.js

```diff
@@ -11,12 +11,17 @@ import DrawerDashBoard from './DrawerDashBoard'
 import LeafletMap from './LeafletMap'
 import LeafletMapControl from './LeafletMapControl'
 import BottomSummaryPanel from './BottomSummaryPanel'
-import { convertQueryParamsToIds, getChoices } from '../lib/array-helpers'
+import {
+  convertQueryParamsToIds,
+  getChoices,
+  getSitesGroupBySampleEvents,
+} from '../lib/array-helpers'
 
 class MermaidDash extends Component {
```

Comment:
why did you choose a class component instead of functional when this was created?

Reply:
It was way back Class component still popular, it used to be all class components before. I have modified the rest and leave the most central part as is. (:) having a mindset where it ain't broken, don't change it)

Reply:
oh, makes sense. I assumed the initial creation of this app was more recent, thats why I asked. :)